src/core/execute.c

```diff
@@ -3690,8 +3717,17 @@ static int exec_child(
         /* In case anything used libc syslog(), close this here, too */
         closelog();
 
-        n_fds = n_socket_fds + n_storage_fds;
-        r = close_remaining_fds(params, runtime, dcreds, user_lookup_fd, socket_fd, params->exec_fd, fds, n_fds);
+        int keep_fds[n_fds + 1];
```

Comment:
hmm are you sure we have enough protection in place that this is not going to grow beyond what the cache is good for?

Reply:
There's an assert that the array is large enough. This function is static and is only called from three fixed places where the size is precalculated. So I think it's OK.